pkcs5/src/pbes2/encryption.rs

```diff
@@ -28,37 +28,51 @@ pub fn encrypt_in_place<'b>(
     password: impl AsRef<[u8]>,
     buffer: &'b mut [u8],
     pos: usize,
-) -> Result<&'b [u8], CryptoError> {
-    let encryption_key = EncryptionKey::derive_from_password(
-        password.as_ref(),
-        &params.kdf,
-        params.encryption.key_size(),
-    )?;
+) -> Result<&'b [u8]> {
+    let key_size = params.encryption.key_size();
+    let algo_params_invalid_error = Error::AlgorithmParametersInvalid {
+        oid: params.encryption.oid(),
+    };
```

Comment:
Rather than eagerly calculating this, you could make it a function on `Parameters`, e.g. `Parameters::invalid_error`.